lib/charms/kubernetes_charm_libraries/v0/multus.py

```diff
@@ -0,0 +1,413 @@
+                    spec={
+                        "config": json.dumps(
+                            {
+                                "cniVersion": "0.3.1",
+                                "type": "macvlan",
+                                "ipam": {"type": "static"},
+                                "capabilities": {"mac": True},
+                            }
+                        )
+                    }
+                ),
+                NetworkAttachmentDefinition(
+                    metadata=ObjectMeta(name="core-net"),
+                    spec={
+                        "config": json.dumps(
+                            {
+                                "cniVersion": "0.3.1",
+                                "type": "macvlan",
+                                "ipam": {"type": "static"},
+                                "capabilities": {"mac": True},
+                            }
+                        )
+                    }
+                ),
+            ],
+            network_annotations=[
+                NetworkAnnotation(
+                    name="access-net",
+                    interface="access",
+                    ips=[self._access_network_ip],
+                ),
+                NetworkAnnotation(
+                    name="core-net",
+                    interface="core",
+                    ips=[self._core_network_ip],
+                ),
+            ],
+        )
+```
+"""
+
+import json
+import logging
+import signal
+import sys
+from dataclasses import asdict, dataclass
+from typing import Optional
+
+import httpx
+from lightkube import Client
+from lightkube.core.exceptions import ApiError
+from lightkube.generic_resource import GenericNamespacedResource, create_namespaced_resource
+from lightkube.models.core_v1 import Capabilities
+from lightkube.resources.apps_v1 import StatefulSet
+from lightkube.types import PatchType
+from ops.charm import CharmBase, EventBase, RemoveEvent
+from ops.framework import Object
+
+# The unique Charmhub library identifier, never change it
+LIBID = "75283550e3474e7b8b5b7724d345e3c2"
+
+# Increment this major API version when introducing breaking changes
+LIBAPI = 0
+
+# Increment this PATCH version before using `charmcraft publish-lib` or reset
+# to 0 if you are raising the major API version
+LIBPATCH = 1
+
+
+logger = logging.getLogger(__name__)
+
+NetworkAttachmentDefinition = create_namespaced_resource(
+    group="k8s.cni.cncf.io",
+    version="v1",
+    kind="NetworkAttachmentDefinition",
+    plural="network-attachment-definitions",
+)
+
+
+@dataclass
+class NetworkAnnotation:
+    """NetworkAnnotation."""
+
+    name: str
+    interface: str
+    ips: Optional[list] = None
+
+    dict = asdict
+
+
+class KubernetesMultusError(Exception):
+    """KubernetesMultusError."""
+
+    def __init__(self, message: str):
+        self.message = message
+        super().__init__(self.message)
+
+
+class Kubernetes:
```

Comment:
isn't the name a bit too general? couldn't it be confused with the `kubernetes-client`?

Reply:
Fixed, using KubernetesClient now